homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,212 @@
+"""RKI Covid numbers sensor."""
+
+from datetime import datetime, timedelta
+import logging
+from typing import Callable, Dict, Optional
+
+from rki_covid_parser.parser import RkiCovidParser
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.components.sensor import PLATFORM_SCHEMA, STATE_CLASS_MEASUREMENT
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import update_coordinator
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import get_coordinator
+from .const import (
+    ATTR_COUNTY,
+    ATTRIBUTION,
+    CONF_BASEURL,
+    CONF_DISTRICT_NAME,
+    CONF_DISTRICTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# wait for x minutes after restart, before refreshing
+SCAN_INTERVAL = timedelta(minutes=10)
+
+# schema for each config entry
+DISTRICT_SCHEMA = vol.Schema({vol.Required(CONF_NAME): cv.string})
+
+# schema for each platform sensor
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_BASEURL): str,
+        vol.Required(CONF_DISTRICTS): vol.All(cv.ensure_list, [DISTRICT_SCHEMA]),
+    }
+)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "weekIncidence": "mdi:clipboard-pulse",
+    "casesPer100k": "mdi:home-group",
+    "newCases": "mdi:shield-bug",
+    "newDeaths": "mdi:shield-cross",
+    "newRecovered": "mdi:shield-sync",
+}
+
+DISTRICT_SENSORS = {
+    "hospitalizationCasesBaby": "mdi:baby-face-outline",
+    "hospitalizationIncidenceBaby": "mdi:baby-face",
+    "hospitalizationCasesChildren": "mdi:account-child-outline",
+    "hospitalizationIncidenceChildren": "mdi:account-child",
+    "hospitalizationCasesTeen": "mdi:face-woman",
+    "hospitalizationIncidenceTeen": "mdi:face-woman-outline",
+    "hospitalizationCasesGrown": "mdi:face-man",
+    "hospitalizationIncidenceGrown": "mdi:face-man-outline",
+    "hospitalizationCasesSenior": "mdi:account-cowboy-hat-outline",
+    "hospitalizationIncidenceSenior": "mdi:account-cowboy-hat",
+    "hospitalizationCasesOld": "mdi:human-white-cane",
+    "hospitalizationIncidenceOld": "mdi:human-cane",
+}
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: Callable,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+) -> None:
+    """Set up the sensor platform."""
+    _LOGGER.debug("setup sensor for platform")
+    session = async_get_clientsession(hass)
+
+    if CONF_BASEURL in config:
+        _LOGGER.warning("Baseurl is not supported anymore.")
```

Comment:
Why do you add this? This is a new integration, thus it doesn't make sense to deprecate a feature that never was supported.

Reply:
The integration was published via HACS before. In an earlier version there was support for a custom base url

Reply:
Why do you use `async_setup_platform` by the way, and not `async_setup_entry`? It seems you didn't fully migrate to a Config Flow with Config Entries.

Reply:
As @gjohansson-ST mentioned in the other comment, I would remove this part since it is not required and should not be added.

+ remove the other YAML parts / deprecation notice

Reply:
I removed all deprecated stuff and support for setup from config file